### ISSUE TITLE
Move point outside field without exiting snippet

### ISSUE THREAD

**Issue by ramblehead** [joaotavora/yasnippet#719]
Is it possible to retain yas snippet open while moving point outside its field in the same buffer? (e.g. to copy some text to yank it into the snippet).

**Comment by joaotavora** [comment]
No. Doing so would violate big assumptions of the snippet expansion/navigation model. Instead, you should mark and save the region beforehand.

We can probably mark this as a feature request, though.

**Comment by ramblehead** [comment]
Thank you for the quick reply.

Currently it is possible to move point to a window showing another buffer (e.g. with windmove-right) without exiting the snippet, copy some text there and return to the window with the template being edited (e.g. windmove-left).

If another window is showing the same buffer at another position moving point there (e.g. with windmove-right) would exit yas snippet. To my senses it feels a bit inconsistent.

Would it be possible to use something like what save-excursion does when point moves out of yas field, and then have an additional interactive function to restore the state as it was just before leaving the field?

**Comment by joaotavora** [comment]
> To my senses it feels a bit inconsistent.

Perhaps. A lot more consistent than a million of active snippets hanging around that you forgot to exit properly though. 

This could be an option, I don't think it's extremely hard to do, but unless the user know what he is doing, really strage things might happen after he/she leaves unfinished snippets.

**Comment by ramblehead** [comment]
I didn’t think about possibility of starting another template in the same buffer before closing the previous one. It does not sound like a good idea indeed.

Can only think of switching buffer to read-only while temporarily leaving yas field...

**Comment by joaotavora** [comment]
> Can only think of switching buffer to read-only while temporarily leaving yas field...

Perhaps. An idea you can explore in a pull-request... 

Here's another: But what if the user changes his/mind and doesn't want to continue the snippet anymore? a read-only buffer (save for the snippet) would be a bummer. So perhaps exit only just before the next change (insertion/deletion) outside the currently active field. There are emacs hooks for this, I think.

**Comment by ramblehead** [comment]
Excellent idea. That would be brilliant if it can be done.

pull-request is tempting :) Haven’t looked into yas code yet – had to check here first if I am not missing something obvious.
